src/ledgerwriter/src/test/java/anthos/samples/financedemo/ledgerwriter/LedgerWriterControllerTest.java

```diff
@@ -79,7 +101,239 @@ void readiness() {
     }
 
     @Test
-    void addTransaction() {
-        // TODO: [issue-52] add tests to addTransaction
+    @DisplayName("Given the transaction routing number is different than the" +
+            "local routing number, return HTTP Status 201")
+    void addTransactionSuccessWhenDiffThanLocalRoutingNum() {
+        // Given
+        when(transaction.getFromRoutingNum()).thenReturn(NON_LOCAL_ROUTING_NUM);
+
+        // When
+        final ResponseEntity actualResult =
+                ledgerWriterController.addTransaction(
+                        BEARER_TOKEN, transaction);
+
+        // Then
+        assertNotNull(actualResult);
+        assertEquals(ledgerWriterController.READINESS_CODE,
+                actualResult.getBody());
+        assertEquals(HttpStatus.CREATED, actualResult.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("Given the transaction routing number is the same as the " +
```

Comment:
`transaction routing number` == `local routing number`, does this mean a user sending payment to self?

Reply:
since it's routing number, I think it just means it's an internal payment rather than external deposit?

nit: it might help to change the text here to something like "Given the transaction is internal and the transaction amount == sender balance..."

Reply:
Updated display names